shm/game.cpp

```diff
@@ -0,0 +1,380 @@
+    std::cout << "/" << std::setfill(' ') << std::setw(100) << "/\n";
+    std::cout << "/" << std::setfill(' ') << std::setw(55) << "SORRY!!!" << std::setw(45) << "/\n";
+    std::cout << "/" << std::setfill(' ') << std::setw(100) << "/\n";
+    std::cout << "/" << std::setfill('-') << std::setw(100) << "/\n";
+}
+
+bool Game::checkWinCondition()
+{
+    if (player_->getMoney() >= finalGoal_)
+    {
+        return true;
+    }
+    return false;
+}
+
+bool Game::checkLoseCondition()
+{
+    return player_->getMoney() == 0 || (time_->getDays() >= gameDayes_ && player_->getMoney() < finalGoal_);
+}
+
+void Game::chooseOption(Game::MenuOption option)
+{
+    switch (option)
+    {
+    case MenuOption::Travel:
+        travel();
+        break;
+    case MenuOption::PrintCargo:
+        printCargo();
+        break;
+    case MenuOption::Buy:
+        buy();
+        break;
+    case MenuOption::Sell:
+        sell();
+        break;
+    case MenuOption::Position:
+        printMap(player_);
+        break;
+    case MenuOption::Exit:
+        exit();
+        break;
+    default:
+        std::cout << "Try again\n";
+        break;
+    }
+}
+
+void Game::exit()
+{
+    std::exit(0);
+}
+
+void Game::sell()
+{
+    std::cout << "Your are selling. Remember that selling will take one day!\n";
+    if (!map_.getIsland(player_->getPlayerPosition()))
+    {
+        std::cout << "Captain, we are on the sea, you can not sell enything!\n";
+        return;
+    }
+    if (player_->getAvailableSpace() == CAPACITY)
+    {
+        std::cout << "Captain, we have nothing to sell!\n";
+        return;
+    }
+    std::cout << "What you want to sell?" << '\n';
+    player_->printCargo();
+    std::cout << "At store prices: \n";
+    auto actualStoreVector = map_.getIsland(player_->getPlayerPosition())->getStore()->getCargoOfStore();
+    std::vector<std::shared_ptr<Cargo>>::iterator atStore;
+    for (auto object : ship_->getCargosVector())
+    {
+        atStore = std::find_if(actualStoreVector.begin(), actualStoreVector.end(), [&object](const auto &atStore)
+                               { return object->getName() == atStore->getName(); });
+        if (atStore != actualStoreVector.end())
+        {
+            std::cout << "Name: " << (*atStore)->getName() << ", Price: " << (*atStore)->getPrice() << '\n';
+        }
+    }
+    std::cout << ship_->getCargosVector().size() + 1 << " Exit\n";
+    int sellOption = 0;
+    std::cin >> sellOption;
+    while (ship_->getCargosVector().size() < sellOption)
+    {
+        if (sellOption == ship_->getCargosVector().size() + 1)
+        {
+            std::cout << "Enough trade for today.\n";
+            return;
+        }
+        std::cout << "Segmentation fault Captain, you have to change your decision!\n";
+        std::cin >> sellOption;
+    }
+    auto objectToSell = ship_->getCargosVector()[sellOption - 1];
```

Comment:
Hmm better will be `getCargoAt(sellOption - 1)`. Also if you don't modify it mark it as const